### PR TITLE
[ONNX] Fix onnx logical functions' dtype

### PR DESCRIPTION
**Pull request by qqaatw**
Part of #79263

Before: output bool is casted back to input's dtype.
After: no longer casted back.

### CODE REVIEW

**Comment by justinchuby** [comment]
@BowenBao could you help reviewing this change?

**Comment by justinchuby** [comment]
@BowenBao could you help reviewing this change?

**Comment by BowenBao** [comment]
@qqaatw are these necessary? It appears current code matches `torch.logical_and` behavior. I wonder what's causing the test failure in the other dtype check PR.

```python
>>> x = torch.randint(10, (5, 5), dtype=torch.int32)
>>> y = torch.randint(10, (5, 5), dtype=torch.int32)
>>> torch.logical_and(x, y).dtype
torch.bool
```

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1415,28 +1415,11 @@ def bitwise_not(g, inp):
...
-                fn(g, input, other),
-                to_i=symbolic_helper.cast_pytorch_to_onnx[to_type],
-            )
-
-        return wrap_with_cast
-
-    return decorator
-
-
 def wrap_logical_op_with_cast_to_and_from(to_type):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1418]: I think this decorator and its usage can all be removed. Casting to bool are automatically handled in `scalar_type_analysis.cpp` already I believe.

Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py:1418]: You are right, I got confused with the `onnx::Greater` that were actually handled in the mentioned pass.
Please rename the decorator, as now it is no longer casting results.

**Review by BowenBao — APPROVED** [review]

**Pull request closed without merging**